source/value-object.js

```diff
@@ -5,13 +5,12 @@ Space.messaging.Serializable.extend(Space.domain, 'ValueObject', {
   },
 
   _hasSameValues: function(other) {
-    var hasSameValues = true;
-    for(var key in this.fields()) {
-      if(this[key] instanceof Space.domain.ValueObject) {
-        if(!this[key].equals(other[key])) hasSameValues = false;
-      }
-      else {
-        if(this[key] !== other[key]) hasSameValues = false;
+    let hasSameValues = true;
```

Comment:
Not necessary since it checks if *any* field is not equal. If this is not the case then they are obviously equal :wink: the other way round is not that easy! You cannot just say at one point "ah ok now i know that they are equal"